charmhelpers/contrib/openstack/utils.py

```diff
@@ -849,7 +849,8 @@ def openstack_upgrade_available(package):
         avail_vers = get_os_version_codename_swift(codename)
     else:
         try:
-            avail_vers = get_os_version_install_source(src)
+            avail_vers = get_os_version_install_source(src,
+                                                       raise_exception=True)
```

Comment:
I've spent a while staring at this, as it was going around my head, and I then realised that this is changing the nature of `openstack_upgrade_availble()` from the existing behaviour of erroring out (using `error_out()` in `get_os_version_codename()`) to a new behaviour of raising an exception (`ValueError`).

To fix this, this  function (`openstack_upgrade_avialble()`) should *also* take an optional `raise_exception=False` parameter, and whatever calls it should pass in `raise_exception=True` to get the new behaviour.  Otherwise, a whole load of charms may get different behaviour from what they are expecting currently.